randovania/games/prime1/exporter/patch_data_factory.py

```diff
@@ -259,7 +338,174 @@ def create_data(self) -> dict:
+                            dst_name = None
+                            dst_dock = None
+                            for (name, dock) in candidates:
+                                if are_rooms_compatible(area.name, None, name, None):
+                                    dst_name = name
+                                    dst_dock = dock
+                                    break
+
+                            # If that wasn't successful, pick random destinations until it works out
+                            while dst_name is None or dst_dock is None or not are_rooms_compatible(area.name, None, dst_name, None):
+                                dst_name = self.rng.choice(world.areas).name
+
+                                if len(area_dock_nums[dst_name]) == 0:
+                                    dst_dock = None
+                                    continue
+
+                                dst_dock = self.rng.choice(area_dock_nums[dst_name])
+
+                            # Don't use this dock as a destination again unless there are no other options
+                            try:
+                                candidates.remove((dst_name, dst_dock))
+                            except ValueError:
+                                # print("re-used %s:%d" % (dst_name, dst_dock))
+                                pass
+                            
+                            used_room_pairings.append({area.name, dst_name})
+
+                            world_data[world.name]["rooms"][area.name]["doors"][str(dock_num)] = {
+                                "destination": {
+                                    "roomName": dst_name,
+                                    "dockNum": dst_dock,
+                                }
+                            }
+                elif self.configuration.room_rando == RoomRandoMode.TWO_WAY:
+                    assert len(candidates) % 2 == 0
+
+                    def next_candidate(max_index):
+                        for (src_name, src_dock) in candidates:
+                            if size_indices[src_name] > max_index:
+                                return (src_name, src_dock)
+                        return (None, None)
+
+                    def pick_random_dst(src_name, src_dock):
+                        for (dst_name, dst_dock) in candidates:
+                            if are_rooms_compatible(src_name, src_dock, dst_name, dst_dock):
+                                return (dst_name, dst_dock)
+                        return (None, None)
+
+                    # Randomly pick room sources, starting with the largest room first, then randomly pick a compatible destination
+                    max_index = 1.01
+                    shuffled = dict()
+                    while len(candidates) != 0:
+                        if max_index < -0.00001:
+                            assert "Failed to find pairings for %s" % str(candidates)
```

Comment:
this assert will always succeed I think? maybe you wanted a `raise Exception()`?